backend/matching-service/src/controllers/handlers/socket-handler.ts

```diff
@@ -0,0 +1,126 @@
+import { Socket } from 'socket.io';
+import { io } from '../../app';
+import RoomManager from '../../lib/utils/roomManager';
+import Preferences from '../../models/types/preferences';
+import Partner from '../../models/types/partner';
+import Room from '../../models/types/room';
+import Logger from '../../lib/utils/logger'
+
+const timeout = Number(process.env.MATCHING_TIMEOUT) || 60000;
+const rm: RoomManager = RoomManager.getInstance();
+
```

Comment:
I think i can put it here first before the mtg but how would u know if the coding session is done? I was thinking of using caching so your room manager emits the roomId to both cache and frontend (for collab-service to use) and then once collab service is done it will remove it from the cache.

Reply:
But yeah I think need to do some research on this cos i have 0 experience in caching but if u have any ideas pls feel free to share